src/System.Net.HttpListener/src/System/Net/HttpListener.cs

```diff
@@ -112,6 +112,158 @@ public HttpListenerPrefixCollection Prefixes
+                        i++;
+                    }
+                }
+                if (NetEventSource.IsEnabled) NetEventSource.Info(this, "mapped uriPrefix:" + uriPrefix + " to registeredPrefix:" + registeredPrefix);
+                if (_state == State.Started)
+                {
+                    if (NetEventSource.IsEnabled) NetEventSource.Info(this, "Calling Interop.HttpApi.HttpAddUrl[ToUrlGroup]");
+                    AddPrefixCore(registeredPrefix);
+                }
+                _uriPrefixes[uriPrefix] = registeredPrefix;
+                _defaultServiceNames.Add(uriPrefix);
+            }
+            catch (Exception exception)
+            {
+                if (NetEventSource.IsEnabled) NetEventSource.Error(this, $"AddPrefix {exception}");
+                throw;
+            }
+            finally
+            {
+                if (NetEventSource.IsEnabled) NetEventSource.Exit(this, $"prefix: {registeredPrefix}");
+            }
+        }
+        
+        internal bool ContainsPrefix(string uriPrefix) => _uriPrefixes.Contains(uriPrefix);
+
+        internal bool RemovePrefix(string uriPrefix)
+        {
+            if (NetEventSource.IsEnabled) NetEventSource.Enter(this, $"uriPrefix: {uriPrefix}");
+            try
+            {
+                CheckDisposed();
+                if (NetEventSource.IsEnabled) NetEventSource.Info(this, $"uriPrefix: {uriPrefix}");
+                if (uriPrefix == null)
+                {
+                    throw new ArgumentNullException(nameof(uriPrefix));
+                }
+
+                if (!_uriPrefixes.Contains(uriPrefix))
+                {
+                    return false;
+                }
+
+                if (_state == State.Started)
+                {
+                    RemovePrefixCore((string)_uriPrefixes[uriPrefix]);
+                }
+
+                _uriPrefixes.Remove(uriPrefix);
+                _defaultServiceNames.Remove(uriPrefix);
+            }
+            catch (Exception exception)
+            {
+                if (NetEventSource.IsEnabled) NetEventSource.Error(this, $"RemovePrefix {exception}");
+                throw;
+            }
+            finally
+            {
+                if (NetEventSource.IsEnabled) NetEventSource.Exit(this, $"uriPrefix: {uriPrefix}");
+            }
+            return true;
+        }
+
+        internal void RemoveAll(bool clear)
+        {
+            if (NetEventSource.IsEnabled) NetEventSource.Enter(this);
+            try
+            {
+                CheckDisposed();
+                // go through the uri list and unregister for each one of them
+                if (_uriPrefixes.Count > 0)
+                {
+                    if (_state == State.Started)
+                    {
+                        foreach (string registeredPrefix in _uriPrefixes.Values)
+                        {
+                            // ignore possible failures
```

Comment:
It was in the old implementation. I think it's something to do with ignoring the `bool` return value of `RemovePrefixCore`, so it's dead